### PR TITLE
Prevent GSSError/_display_status() infinite recursion

### PR DESCRIPTION
**Pull request by frozencemetery**
I was unable to reproduce the problem, but this should prevent the issue (which is #111).

### CODE REVIEW

**Comment by frozencemetery** [comment]
(Pulled some of the prep commits out of #109)

**Comment by tiran** [comment]
I'm not a fan of assert() for flow control. Some flags (e.g. ```-O```) remove assert statements.

**Comment by frozencemetery** [comment]
Christian Heimes <notifications@github.com> writes:

> I'm not a fan of assert() for flow control. Some flags (e.g. ```-O```)
> remove assert statements.

Understood.  Do you have a recommendation instead?  My original thought
was to raise an exception manually, and I only switched to
assert() because the assertion failure exception looked closest to the
kind of problem that was occurring.

**Comment by tiran** [comment]
Any other exception will do. I like ```ValueError``` for an invalid value. You can subclass ValueError if you prefer to catch just that specific exception.

**Comment by frozencemetery** [comment]
Switched to `ValueError`.  (Commit order changed too, but that's not important.)

File: gssapi/raw/misc.pyx [hunk 1]
```diff
@@ -165,13 +165,15 @@ def _display_status(unsigned int error_code, bint is_major_code,
...
         gss_release_buffer(&min_stat, &msg_buff)
         return (msg_out, msg_ctx_out, call_again)
     else:
-        # NB(directxman12): this is highly unlikely to cause a recursive loop
-        raise GSSError(maj_stat, min_stat)
+        # This hides whatever error gss_display_status is complaining about,
+        # but obviates infinite recursion into stack exhaustion.  The
+        # exception raised here is handled by get_all_statuses(), which prints
+        # the code.
+        raise ValueError("gss_display_status call returned failure.")
```
Comment (DirectXMan12) [inline gssapi/raw/misc.pyx]: please attach the error codes from `gss_display_status` into this message

**Comment by frozencemetery** [comment]
Updated to include `gss_display_status` errors (for future debugging) as per request.

**Pull request merged**